oneview/resource_server_hardware.go

```diff
@@ -261,14 +279,28 @@ func resourceServerHardwareUpdate(d *schema.ResourceData, meta interface{}) erro
 			return err
 		}
 	}
-	if d.HasChange("username") || d.HasChange("password") || d.HasChange("configuration_state") {
-		return errors.New("Fields like username, password and configuration_state cannot be changed")
+	if d.HasChange("username") || d.HasChange("password") || d.HasChange("configuration_state") || d.HasChange("initial_scope_uris") {
+		return errors.New("Fields like username, password, configuration_state and initial_scope_uris cannot be changed")
 	}
 	d.SetId(d.Id())
 
 	return resourceServerHardwareRead(d, meta)
 }
 
 func resourceServerHardwareDelete(d *schema.ResourceData, meta interface{}) error {
-	return errors.New("Server hardware delete Operation is not supported.")
+	config := meta.(*Config)
+
+	hardwareType, err := config.ovClient.GetServerHardwareTypeByUri(utils.Nstring(d.Get("server_hardware_type_uri").(string)))
+	if err != nil {
+		return err
+	}
+	if hardwareType.Platform == "RackServer" {
+		err := config.ovClient.DeleteServerHardware(utils.Nstring(d.Get("uri").(string)))
+		if err != nil {
+			return err
+		}
+	} else {
+		return errors.New("Server hardware delete is not supported")
```

Comment:
This error message can be little more descriptive like Deletion of Server hardware is only supported for Rack Servers or something like this